spec/features/user_groups_spec.rb

```diff
@@ -28,4 +28,35 @@
 
     expect(page).to have_text(group.name)
   end
+
+  scenario "user can leave a group they are part of" do
+    user = create(:user)
+    group = create(:group)
+
+    sign_in(user)
+    group.users << user
+    visit dashboard_path
+    click_link group.name
+    click_link "Leave Group"
+
+    expect(page).to have_text("Join Group")
+    expect(user.member?(group)).to eq false
+
+    group.remove_member(user)
+    expect(group.users.length).to eq 0
+    expect(group.users).not_to include user
```

Comment:
@tabfugnic this test is confusing me. So this test passes simple because of line 45. line 40 click_link "Leave Group'' calls group.remove in the MembershipController::destroy. This unit test for this method works as well. however also when run through this manually it seems to work fine.